asset/asset.go

```diff
@@ -943,6 +944,447 @@ type GroupKeyReveal interface {
 	GroupPubKey(assetID ID) (*btcec.PublicKey, error)
 }
 
+// NewNonSpendableScriptLeaf creates a new non-spendable tapscript script leaf
+// that includes the specified data. If the data is nil, the leaf will not
+// contain any data but will still be a valid non-spendable script leaf.
+//
+// The script leaf is made non-spendable by including an OP_RETURN opcode at the
+// start of the script. While the script can still be executed, it will always
+// fail and cannot be used to spend funds.
+func NewNonSpendableScriptLeaf(data []byte) lfn.Result[txscript.TapLeaf] {
+	// Construct a script builder and add the OP_RETURN opcode to the start
+	// of the script to ensure that the script is non-executable.
+	scriptBuilder := txscript.NewScriptBuilder().AddOp(txscript.OP_RETURN)
+
+	// Add the data to the script if it is provided.
+	if data != nil {
+		scriptBuilder.AddData(data)
+	}
+
+	// Construct script from the script builder.
+	script, err := scriptBuilder.Script()
+	if err != nil {
+		return lfn.Errf[txscript.TapLeaf]("failed to construct "+
+			"non-spendable script: %w", err)
+	}
+
+	// Create a new tapscript leaf from the script.
+	leaf := txscript.NewBaseTapLeaf(script)
+	return lfn.Ok(leaf)
+}
+
+// TLV types for GroupKeyReveal.
+var (
```

Comment:
So you can just make these type aliases instead of make a new pointer. 

Here's an example from `lnd`: https://github.com/lightningnetwork/lnd/blob/8ecef03315f749ae8820b4735978ca2e0552de2e/lnwire/musig2.go#L10-L15

Reply:
I've gone with:
```
type GroupKeyRevealTlvType = tlv.Type

const (
	GKRVersion           GroupKeyRevealTlvType = 0
	GKRInternalKey       GroupKeyRevealTlvType = 2
	GKRTapscriptRoot     GroupKeyRevealTlvType = 4
	GKRCustomSubtreeRoot GroupKeyRevealTlvType = 7
)
```
for now because I've seen that pattern used quite a bit in the old TLV setups (before `tlv.RecordT`).